cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java

```diff
@@ -117,53 +117,46 @@ public void readLog(Location file, Filter logFilter, long fromTimeMs, long toTim
           return ImmutableList.of();
         }
 
-        GenericRecord datum;
         List<List<LogEvent>> logSegments = Lists.newArrayList();
         int count = 0;
 
         // Calculate skipLen based on fileLength
-        long skipLen = file.length() / 10;
+        long length = file.length();
+        LOG.trace("Got file Length {}", length);
+        long skipLen = length / 10;
         if (skipLen > DEFAULT_SKIP_LEN) {
           skipLen = DEFAULT_SKIP_LEN;
         } else if (skipLen <= 0) {
           skipLen = DEFAULT_SKIP_LEN;
         }
 
+        // For open file, start from file length and read till the actual eof
+        dataFileReader.sync(length);
+        long finalSync = dataFileReader.tell();
         List<LogEvent> logSegment = Lists.newArrayList();
+        logSegment = readToNextSync(logFilter, fromTimeMs, dataFileReader, -1, -1);
```

Comment:
That it added [here](https://github.com/caskdata/cdap/pull/5334/commits/4edfab4b5bc62dbd2edef991e0ab44bae1e3118f#diff-8855c67680faf144d9fa02d054d7e5deR145)